src/components/Pressable.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { GestureObjects as Gesture } from '../handlers/gestures/gestureObjects';
+import { GestureDetector } from '../handlers/gestures/GestureDetector';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  TouchData,
+} from '../handlers/gestureHandlerCommon';
+import { PressEvent, PressableEvent, PressableProps } from './PressableProps';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+import { LongPressGestureHandlerEventPayload } from '../handlers/LongPressGestureHandler';
+import { Insets, StyleProp, View, ViewStyle } from 'react-native';
+import RNButton from './GestureHandlerButton';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+const DEFAULT_HOVER_DELAY = 0;
+
+const numberAsInset = (value: number): Insets => ({
+  left: value,
+  right: value,
+  top: value,
+  bottom: value,
+});
+
+const addInsets = (a: Insets, b: Insets): Insets => ({
+  left: (a.left ?? 0) + (b.left ?? 0),
+  right: (a.right ?? 0) + (b.right ?? 0),
+  top: (a.top ?? 0) + (b.top ?? 0),
+  bottom: (a.bottom ?? 0) + (b.bottom ?? 0),
+});
+
+const touchToPressEvent = (data: TouchData, timestamp: number): PressEvent => ({
+  identifier: data.id,
+  locationX: data.x,
+  locationY: data.y,
+  pageX: data.absoluteX,
+  pageY: data.absoluteY,
+  target: 0, // fixme if possible, set to correct target ID
+  timestamp: timestamp,
+  touches: [], // intentionally empty
+  changedTouches: [], // intentionally empty
+});
+
+const changeToTouchData = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): TouchData => ({
+  id: 0, // fixme if possible, set to correct pointer ID
+  x: event.x,
+  y: event.y,
+  absoluteX: event.absoluteX,
+  absoluteY: event.absoluteY,
+});
+
+const isTouchWithinInset = (
+  // all of these are guaranteed to be their type,
+  // but we cannot use '!' anymore
+  touch?: TouchData,
+  inset?: Insets,
+  dimensions?: { width: number; height: number }
+) =>
+  // ^ to right, ^ to bottom
+  (touch?.x ?? 0) < (inset?.right ?? 0) + (dimensions?.width ?? 0) &&
+  (touch?.y ?? 0) < (inset?.bottom ?? 0) + (dimensions?.height ?? 0) &&
+  (touch?.x ?? 0) > -(inset?.left ?? 0) &&
+  (touch?.y ?? 0) > -(inset?.top ?? 0);
+
+const adaptStateChangeEvent = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): PressableEvent => {
+  const timestamp = Date.now();
+
+  const touchData = changeToTouchData(event);
+
+  const pressEvent = touchToPressEvent(touchData, timestamp);
+
+  return {
+    nativeEvent: {
+      touches: [pressEvent],
+      changedTouches: [pressEvent],
+      identifier: event.handlerTag,
+      locationX: event.x ?? -1,
+      locationY: event.y ?? -1,
+      pageX: event.absoluteX ?? -1,
+      pageY: event.absoluteY ?? -1,
```

Comment:
Shouldn't those be defined on the event?

Reply:
I'm not sure 100% if it's defined on every platform, will have to double check it.

Reply:
Addressed this problem in depth in a different comment further up.